File: Game/src/main/java/net/winternetwork/bedwars/game/modules/shop/command/ShopCommands.java
package net.winternetwork.bedwars.game.modules.shop.command;

import lombok.Getter;
import me.saiintbrisson.commands.Execution;
import me.saiintbrisson.commands.annotations.Command;
import net.winternetwork.bedwars.game.modules.Modules;
import net.winternetwork.bedwars.game.modules.shop.ShopModule;
import net.winternetwork.bedwars.game.modules.shop.controller.ShopController;
import net.winternetwork.bedwars.game.modules.shop.model.VillagerModel;
import net.winternetwork.bedwars.game.modules.shop.model.menu.MenuView;
import org.bukkit.Location;
import org.bukkit.entity.Villager;

public class ShopCommands {

    private final ShopController controller = Modules.getModule(ShopModule.class).getShopController();

    @Getter(lazy = true)
    private final MenuView view = Modules.getModule(ShopModule.class).getView();

    @Command(name = "summonshop", permission = "summonshop.use")
    public void summon(Execution execution, String id) {
        Location location = execution.getPlayer().getLocation();

        Villager villager = VillagerModel.spawnVillager(location);

        controller.add(id, villager);
    }

    @Command(name = "openshop", permission = "openshop.use")
    public void openShop(Execution execution) {
        getView().showInventory(execution.getPlayer());
    }
}

File: Game/src/main/kotlin/net/winternetwork/bedwars/game/modules/shop/command/ShopCommands.kt
package net.winternetwork.bedwars.game.modules.shop.command

import me.saiintbrisson.commands.Execution
import me.saiintbrisson.commands.annotations.Command
import net.winternetwork.bedwars.api.util.inject
import net.winternetwork.bedwars.game.modules.shop.ShopController
import net.winternetwork.bedwars.game.modules.shop.ShopModule
import net.winternetwork.bedwars.game.modules.shop.model.VillagerModel
import net.winternetwork.bedwars.game.modules.shop.model.menu.MenuView

class ShopCommands {

    val module: ShopModule by inject()

    private val controller: ShopController by lazy { module.controller }
    private val view: MenuView by lazy { module.view }

    @Command(name = "summonshop", permission = "summonshop.use")
    fun summon(execution: Execution, id: String) {
        val location = execution.player.location

        controller.add(
                id,
                VillagerModel.spawnVillager(location)
        )
    }

    @Command(name = "openshop", permission = "openshop.use")
    fun openShop(execution: Execution) {
        view.showInventory(execution.player)
    }
}